### PR TITLE
Refactor Ingress Configuration based on Processor Type

### LINKED ISSUE #354: Ingress configuration always contains VEBA UI endpoints
**Describe the bug**
It looks like [setup-09-ingress.sh](https://github.com/vmware-samples/vcenter-event-broker-appliance/blob/development/files/setup-09-ingress.sh) does not verify whether the deployment will setup VEBA UI and hence the ingress configuration always contains the VEBA UI endpoints. This is problematic since the behavior for the customer is that VEBA UI configurations were not applied and hence, the system should not configure the VEBA UI endpoints within the ingress controller as this will cause it to have invalid state

**To Reproduce**
1) Deploy OVA using Embedded Knative option BUT do NOT fill in username/password for registering the VEBA UI
2) Endpoints such as /status, /stats and /events fail due to invalid ingress configuration

**Expected behavior**
Ingress configuration does not contain VEBA UI endpoints when user 

**Screenshots**
```
 conditions:
  - errors:
    - message: 'Spec.Routes unresolved service reference: service "vmware-system/veba-ui"
        not found'
      reason: ServiceUnresolvedReference
      status:
```

**Version (please complete the following information):**
 - VEBA Form Factor: Appliance
 - VEBA Version: v0.6.0

**Additional context**
Slack Discussion: https://vmwarecode.slack.com/archives/CQLT9B5AA/p1619001864106000

### PR DESCRIPTION
**Pull request by lamw**
Signed-off-by: William Lam <wlam@vmware.com>

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

Closes #354 

## Testing Verification

* Verified Knative Embedded with and without VEBA UI configuration
* Verified OpenFaaS and EventBridge Processor deployments didn't regress

## Additional Information

N/A

### CODE REVIEW

File: files/knative-external-ingressroute-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  annotations:
+    kubernetes.io/ingress.class: contour-external
```
Comment (embano1) [inline files/knative-external-ingressroute-gateway.yaml:5]: Is this required in external mode?

**Review by embano1 — COMMENTED** [review]
This asks for templating :) but I'm ok right now to reduce the scope, but it might come back to us at a later point when we want to make changes to the Ingress configuration.

**Comment by lamw** [comment]
Like YTT or something else? Sorry, so you have example or reference?

File: files/knative-external-ingressroute-gateway.yaml (hunk 1 above)
Comment (lamw) [reply files/knative-external-ingressroute-gateway.yaml]: Good catch! No, its not needed. Will fix

**Review by embano1 — APPROVED** [review]
LGTM

**Pull request merged**